bindings/nodejs/examples/how_tos/alias/governance-transition.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    AliasOutput,
+    StateControllerAddressUnlockCondition,
+    UnlockConditionType,
+    Utils,
+    Wallet,
+    initLogger,
+} from '@iota/sdk';
+
+// This example uses secrets in environment variables for simplicity which should not be done in production.
+require('dotenv').config({ path: '.env' });
+
+// Run with command:
+// yarn run-example ./how_tos/alias/governance-transition.ts
+
+// In this example we will update the state controller of an alias output.
+async function run() {
+    initLogger();
+    if (!process.env.FAUCET_URL) {
+        throw new Error('.env FAUCET_URL is undefined, see .env.example');
+    }
+    if (!process.env.WALLET_DB_PATH) {
+        throw new Error('.env WALLET_DB_PATH is undefined, see .env.example');
+    }
+    if (!process.env.STRONGHOLD_PASSWORD) {
+        throw new Error(
+            '.env STRONGHOLD_PASSWORD is undefined, see .env.example',
+        );
+    }
+    try {
+        // Create the wallet
+        const wallet = new Wallet({
+            storagePath: process.env.WALLET_DB_PATH,
+        });
+
+        // Get the account we generated with `01-create-wallet`
+        const account = await wallet.getAccount('Alice');
+
+        // May want to ensure the account is synced before sending a transaction.
+        const balance = await account.sync();
+
+        if (balance.aliases.length == 0) {
+            throw new Error(`No Alias available in account 'Alice'`);
+        }
+
+        // We try to update the first alias in the account
+        const aliasId = balance.aliases[0];
+
+        const aliasOutputData = (
+            await account.unspentOutputs({ aliasIds: [aliasId] })
+        )[0];
+        console.log(
+            `Alias ${aliasId} found in unspent output: ${aliasOutputData.outputId}`,
+        );
+
+        await wallet.setStrongholdPassword(process.env.STRONGHOLD_PASSWORD);
+
+        const newStateController = Utils.parseBech32Address(
+            (await account.generateEd25519Addresses(1))[0].address,
+        );
+
+        const aliasOutput = aliasOutputData.output as AliasOutput;
+        const updatedUnlockConditions = aliasOutput.unlockConditions.map(
+            (unlock) => {
+                if (unlock.type == UnlockConditionType.StateControllerAddress) {
+                    return new StateControllerAddressUnlockCondition(
+                        newStateController,
+                    );
+                } else {
+                    return unlock;
+                }
+            },
+        );
+
+        const updatedAliasOutput = await (
+            await wallet.getClient()
+        ).buildAliasOutput({
+            aliasId,
```

Comment:
Do we need to check for null here like in rust?

Reply:
No, since we get the id from the balance it's already not null
In the Rust example it's actually also not needed, but thought it could still be interesting to have in an example in case the id is not returned from the balance, but another output

Reply:
Hmm, well imo they should match so maybe just add a comment about it instead?

Reply:
Just removed it, because I don't know where and how to put it without it being more confusing for users, since in this example it's not needed